### DISCUSSION TITLE
Improve the UTM Builder and link to it from the footer

### DISCUSSION THREAD

**Discussion by derpycoder** [plausible/analytics#718] [Anything Else]
Hey Guys,

I found a way to give back to this community.

I made a small tool to ease creation of UTM Params, hope it's up to your standards.

Checkout my blog post about it: [Derpy UTM Builder: A Campaign URL Builder made with Svelte, Tailwind & Snowpack](https://www.derpycoder.com/derpy-utm-builder-a-campaign-url-builder-made-with-svelte-tailwind-snowpack/?utm_campaign=Derpy+Tools&utm_source=Plausible&utm_medium=Forum).

**Comment by metmarkosaric** [comment]
thanks @abhijit-kar! what do the formats such as "plausible+case" do?

**Reply by derpycoder** [reply]
Hey @metmarkosaric,

It enforces the format you have mentioned in your [blog post](https://plausible.io/blog/utm-tracking-tags#you-cannot-use-spaces-in-utm-tags), so users can maintain consistency in their UTM Params.

Try enabling it, and then type into different fields normally with spacebar. (Try switching between formats once you have filled the page with values!)

That's what I have, turned on, in my blog post as well. As you can see in the screenshot:

[![Derpy UTM Builder](https://user-images.githubusercontent.com/25662120/107905328-5f424480-6f74-11eb-987d-62cac9096966.png)](https://www.abhijit-kar.com/derpy-utm-builder/?utm_campaign=Derpy+Tools&utm_source=Plausible&utm_medium=Forum)

**Reply by metmarkosaric** [reply]
Hi @abhijit-kar! I wonder if there's a way for us to embed your UTM builder (the Plausible part only) directly into our blog post? We send all the UTM tags queries to [this blog post](https://plausible.io/blog/utm-tracking-tags) so could be cool to have the UTM builder there for easy way to create your own links. Paste the original URL and then options to add medium, source and campaign. Then get easy complete link to copy and paste. Thanks!

**Reply by derpycoder** [reply]
Hey Marko,

You can embed the utm builder, as is, directly in an iframe.

```
<iframe src="https://www.abhijit-kar.com/derpy-utm-builder/" title="Derpy UTM Builder" style="width: 100%; height: 100vh; margin-bottom: 30px;" frameborder="0" loading="lazy" allowfullscreen="true">
  <a href="https://www.abhijit-kar.com/derpy-utm-builder/">Derpy UTM Builder</a>
</iframe>
```

It's open source, so you can do whatever you want with it as well. [Derpy UTM Builder Repo](https://github.com/abhijit-kar/derpy-utm-builder)

Or I can modify the code and remove all the fluff and you can host it.

Let me know what you would prefer.

**Reply by metmarkosaric** [reply]
thanks @abhijit-kar! for our page the simplest version would be best. paste the url and then insert source, medium, campaign (not all 3 are required) and then you get the final url.

**Reply by derpycoder** [reply]
Hey Marko,

I will strip it down to bare minimum and host it separately.

Give me a few days.

**Reply by metmarkosaric** [reply]
nice one @abhijit-kar, thank you!

**Reply by derpycoder** [reply]
Hey @metmarkosaric,

I just published the changes.
It doesn't have any of the fluff, and can be directly embedded into the blog post as an iframe.

[Plausible UTM Builder](https://www.abhijit-kar.com/plausible-utm-builder/)

![image](https://user-images.githubusercontent.com/25662120/109137726-8b817080-777f-11eb-8cd6-18cdad40b071.png)


P.S. The whole code fit a single page now: [See VSCode Online](https://github1s.com/abhijit-kar/plausible-utm-builder/blob/HEAD/src/App.svelte).

[GitHub Repo](https://github.com/abhijit-kar/plausible-utm-builder)

**Reply by metmarkosaric** [reply]
Looks beautiful. thanks! Let me try and see if I can figure out how to embed it into that post now

**Reply by metmarkosaric** [reply]
I managed to embed it! And it works too! https://plausible.io/blog/utm-tracking-tags#plausible-utm-tag-builder-tool

Is there a way to remove the bar on the right hand side? And also to shorten the top a bit as there's a lot of white space before the actual builder starts? Thanks again @abhijit-kar!

![Screenshot from 2021-02-25 13-03-25](https://user-images.githubusercontent.com/34340819/109151007-e9a35900-7769-11eb-9bcf-303c52b9af74.png)

**Reply by derpycoder** [reply]
Hi @metmarkosaric,

I removed the min height constraint.

You just have to update the html embed to this:

```
<iframe src="https://www.abhijit-kar.com/plausible-utm-builder/" title="Plausible UTM Tag Builder by DerpyCoder" style="width: 100%;height: 335px;" frameborder="0" loading="lazy" allowfullscreen="true"></iframe>
```

## Result

![image](https://user-images.githubusercontent.com/25662120/109261243-90006480-7825-11eb-9428-66481bb320cc.png)

P.S. Thank You for the mention, I appreciate it.

**Reply by metmarkosaric** [reply]
thank you! looks great! firefox has the scroll bar now which allows you to scroll a tiny bit but not sure if there's a way to fix that

![Screenshot from 2021-02-26 07-34-25](https://user-images.githubusercontent.com/34340819/109264455-4d2e9480-7805-11eb-9712-9996c99ee828.png)

**Reply by derpycoder** [reply]
Hey @metmarkosaric,

I will check on firefox tonight, and make sure the scrollbar doesn't appear.

Till then, if you can try increasing the height in iFrame, it would be awesome.

Scrollbar is appearing because there's more content, but the height is fixed on iFrame to 335px.

**Reply by metmarkosaric** [reply]
thanks! i've played a bit and at 360px it seems better. at least no way to scroll it anymore on neither firefox nor chromium. firefox still shows some kind of bar on the right side but chromium is all fine.

**Reply by derpycoder** [reply]
Hey @metmarkosaric,

I fixed it for firefox, so you won't have to see that abomination of a scrolltrack.

The Plausible blog looks pristine again.

P.S. I didn't have Firefox installed, I should have tested in that browser, my bad.

**Reply by metmarkosaric** [reply]
now it looks perfect! thanks for all the help!

**Reply by metmarkosaric** [reply]
hi @abhijit-kar! we recently introduced support for utm_term and utm_content so not sure if you're available to add them as optional tags in your utm builder embed tool? those are the final utm tags so should be no further changes to this. thanks for your help!

**Comment by metmarkosaric** [comment]
we frequently link people to our [utm tag/utm builder](https://plausible.io/blog/utm-tracking-tags) post. the builder itself currently doesn't support utm_term and utm_content which causes confusion whether Plausible supports those (we do). would be nice if we could have a Plausible utm builder in there that features all tags

**Reply by metmarkosaric** [reply]
and link it somewhere from the footer so it's easier to discover

([idea from twitter](https://twitter.com/JonathanDHolden/status/1561582100633255941))
